gpkit/constraints/set.py

```diff
@@ -177,9 +180,9 @@ def as_gpconstr(self, x0):
         """Returns GPConstraint approximating this constraint at x0
 
         When x0 is none, may return a default guess."""
-        cs = ConstraintSet([constr.as_gpconstr(x0) for constr in self])
-        cs.substitutions.update(self.substitutions)
-        return cs
+        gpconstrs = [constr.as_gpconstr(x0) for constr in self]
+        return ConstraintSet(gpconstrs,
+                             self.substitutions, recursesubs=False)
```

Comment:
This just assumes that the individual constraint's substitutions don't change between SP iterations, because they've all been set to point to the model's substitutions dict; the user / any constraints are welcome to change `self.substitutions`, and it'll be automatically reflected globally.